mlflow/tracing/provider.py

```diff
@@ -78,6 +81,10 @@ def _setup_tracer_provider(disabled=False):
         return
 
     if is_in_databricks_model_serving_environment():
+        if not enable_mlflow_tracing():
```

Comment:
Should we also update https://github.com/mlflow/mlflow/blob/89c44d9e2468cc083a7fd6b3cb6d9921982fd6aa/mlflow/langchain/__init__.py#L638-L639 not to inject callback when tracing is disabled?